### PR TITLE
Allow to override trigger_build from demo project

### PR DESCRIPTION
**Pull request by stsewd**
This is to fix a case in the corporate site

### CODE REVIEW

File: readthedocs/projects/views/private.py [hunk 1]
```diff
@@ -381,6 +379,9 @@ def get_form_kwargs(self):
         """Form kwargs passed in during instantiation."""
         return {'user': self.request.user}
 
+    def trigger_build(self, project):
```
Comment (humitos) [inline readthedocs/projects/views/private.py]: This should probably be called `trigger_initial_build` to keep compatibility naming with the normal import.

Comment (humitos) [reply readthedocs/projects/views/private.py]: Also, please add a docstring here explaining why this is needed to be in a method.

Comment (stsewd) [reply readthedocs/projects/views/private.py]: Not really sure what to put in a one line function

Comment (humitos) [reply readthedocs/projects/views/private.py]: It does not matter too much how many lines it has. Explain _why_ its needed to be override this method here, for example?

Consider that another person is reading this method in 2 months and their will ask themselves: "Why do we have this method here that just call `trigger_build`? Why not just call `trigger_build` it directly instead?". Your docstring should be answer those questions in a clear way.

Comment (stsewd) [reply readthedocs/projects/views/private.py]: I don't want to mention the corporate site here, so not really sure what to put

Comment (humitos) [reply readthedocs/projects/views/private.py]: Don't use the "corporate site" words ;)

"Extend from outside" or whatever that works for you and makes reference the goal of the override: allows to be extensible and change behavior.

**Pull request merged**